File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cs
public class Solution {
    public string MergeAlternately(string word1, string word2) {
        var sb = new StringBuilder();
        int p1 = 0, p2 = 0;
        while (p1 < word1.Length && p2 < word2.Length) {
            Console.WriteLine($"{p1} - {p2}");
            sb.Append(word1[p1++]);
            sb.Append(word2[p2++]);
            
        }

        while (p1 < word1.Length) sb.Append(word1[p1++]);
        while (p2 < word2.Length) sb.Append(word2[p2++]);

        return sb.ToString();
    }
}

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l, r = 0, 0
        ans = []
        while l < len(word1) and r < len(word2):
            ans += word1[l]
            ans += word2[r]
            l += 1
            r += 1
        
        if l < len(word1):
            ans += word1[l:]
        elif r < len(word2):
            ans += word2[r:]
        

        return "".join(ans)